tools/amispec/src/ebs.rs

```diff
@@ -0,0 +1,468 @@
+//! This module implements serde-based (de-)serialization for EBS block devices.
+//!
+//! Performs client-side validation of inputs (e.g. properly checks that invalid fields aren't
+//! enabled based on volume type, or that requested volume sizes are within bounds.)
+use aws_sdk_ec2::types::builders::EbsBlockDeviceBuilder;
+use aws_sdk_ec2::types::{EbsBlockDevice as SdkEbsBlockDevice, VolumeType as SdkVolumeType};
+use bon::Builder;
+use bounded_integer::BoundedU64;
+use num_traits::AsPrimitive;
+use serde::{Deserialize, Serialize};
+use serde_templated::TemplateOf;
+use serde_templated::Templated;
+use snafu::{OptionExt, Snafu};
+use std::fmt::Debug;
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "kebab-case", tag = "volume-type")]
+pub enum EbsBlockDevice {
+    Standard(Standard),
+    Io1(Io1),
+    Io2(Io2),
+    Gp2(Gp2),
+    Gp3(Gp3),
+    St1(St1),
+    Sc1(Sc1),
+}
+
+impl EbsBlockDevice {
+    pub(crate) fn create_sdk_ebs_block_device(&self) -> SdkEbsBlockDevice {
+        let builder = EbsBlockDeviceBuilder::default();
```

Comment:
I'd personally add a macro to cut down the repetition but I know that's not to everyone's taste 😀 
```suggestion
        let builder = EbsBlockDeviceBuilder::default();
        macro_rules! ebs_volume {
            ($device:expr, $voltype:ident) => {
                $device.configure_ebs_block_device(builder).volume_type(SdkVolumeType::$voltype)
            };
        }
```